command/fs.go

```diff
@@ -29,6 +30,12 @@ func (f *FSCommand) Run(args []string) int {
 func getRandomJobAlloc(client *api.Client, jobID string) (string, error) {
 	var runningAllocs []*api.AllocationListStub
 	allocs, _, err := client.Jobs().Allocations(jobID, nil)
+
+	// Check that the job actually has allocations
+	if len(allocs) == 0 {
+		return "", fmt.Errorf("job %q doesn't exist or it has no allocations", jobID)
```

Comment:
I just tested it and I think the err is in fact nil if the job doesn't exist.... But I will see if I can specialize it. I also just noticed that stat and cat will happily continue even with the returned error, so I'll fix that too.